### PR TITLE
Re-created a GitHub build & test pipeline

### PR DESCRIPTION
**Pull request by jstedfast**
### Description of Change ###

Re-created a GitHub build & test pipeline
* Updated UnitTests.csproj to use the latest MSTest.Adaptor/Framework nugets
* Use `dotnet test` to run the unit tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      run: msbuild src\BehaviorsSdk.sln /p:Configuration=Release /p:SignType=real
+    
+    - name: Run Tests
+      run: dotnet test Test\UnitTests\UnitTests.csproj
+
+    - name: Pack Behaviors
+      if: success()
+      run: msbuild src\Microsoft.Xaml.Behaviors\Microsoft.Xaml.Behaviors.csproj /p:Configuration=Release -t:pack
+    
+    - name: Upload NuGet Artifacts
```
Comment (mgoertz-msft) [inline .github/workflows/main.yml:39]: Should we remove the Upload step now that we have this in the Azure Pipeline?

**Review by mgoertz-msft — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (jstedfast) [reply .github/workflows/main.yml:39]: I was wondering about that too

Comment (mgoertz-msft) [reply .github/workflows/main.yml:39]: Never mind. This isn't uploading anything to nuget.org but just keeps these build artifacts for GH workflow runs, so we should keep it.

Comment (mgoertz-msft) [reply .github/workflows/main.yml:39]: From GitHub Copilot :-)

> GitHub stores artifacts in the "Actions" tab of your repository. Here's how you can view them:
> 
> Navigate to the Repository: Go to the repository where the workflow runs.
> 
> Access the Actions Tab: Click on the "Actions" tab at the top of the repository page.
> 
> Select a Workflow Run: Find and click on the workflow run that generated the artifacts. You can identify it by the workflow name and the date it ran.
> 
> View Artifacts: Once you're on the workflow run page, scroll down to the "Artifacts" section. Here you will see a list of artifacts that were uploaded during the workflow run. You can download them from this section.
> 
> You can also access them directly from the workflow run summary page.

**Pull request merged**